pkg/apis/management.cattle.io/v3/pod_security_types.go

```diff
@@ -0,0 +1,92 @@
+package v3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const PodSecurityAdmissionConfigurationTemplateKind = "PodSecurityAdmissionConfigurationTemplate"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PodSecurityAdmissionConfigurationTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Configuration PodSecurityAdmissionConfigurationTemplateSpec `json:"configuration"`
+}
+
+type PodSecurityAdmissionConfigurationTemplateSpec struct {
+	Defaults   PodSecurityAdmissionConfigurationTemplateDefaults   `json:"defaults"`
+	Exemptions PodSecurityAdmissionConfigurationTemplateExemptions `json:"exemptions,omitempty"`
+}
+
+// PodSecurityAdmissionConfigurationTemplateDefaults are applied when a mode label is not set.
+//
+// The level label values must be one of
+// 'privileged' (default), 'baseline', or 'restricted'
+//
+// The version levels must be either 'latest' (default), or a specific version (e.g. 'v1.25')
+type PodSecurityAdmissionConfigurationTemplateDefaults struct {
+	Enforce        string `json:"enforce"`
+	EnforceVersion string `json:"enforce-version" yaml:"enforce-version"`
+	Audit          string `json:"audit"`
+	AuditVersion   string `json:"audit-version" yaml:"audit-version"`
+	Warn           string `json:"warn"`
+	WarnVersion    string `json:"warn-version" yaml:"warn-version"`
+}
+
+type PodSecurityAdmissionConfigurationTemplateExemptions struct {
+	Usernames      []string `json:"usernames"`
+	RuntimeClasses []string `json:"runtimeClasses"`
+	Namespaces     []string `json:"namespaces"`
+}
+
+func NewPodSecurityAdmissionConfigurationTemplateRestricted() PodSecurityAdmissionConfigurationTemplate {
+	return PodSecurityAdmissionConfigurationTemplate{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
Do we have to pass `TypeMeta` here? (Trying to avoid hardcoding `APIVersion`)
For comparison, it's skipped for PSPTs: https://github.com/rancher/rancher/blob/60ab459ac61cae96dfc3145e70e5d3afb9fdf7c8/pkg/data/management/podsecuritypolicytemplate_data.go#L18

Reply:
I'll follow the convention used by PSPTS, wasn't aware we wanted to avoid hard-coding the `APIVersion`

Reply:
I've removed the `TypeMeta` and added the two default templates to `add.go`, they now appear when using `kubectl`